### PR TITLE
Admin assignments

### PR DESCRIPTION
**Pull request by Advayp**
Author: Advay

## What changes were made?
Add functionality for assigning admins to reports.

## Why are these changes important/necessary?
Improve moderation

### CODE REVIEW

File: src/components/admin/ReportView.tsx [hunk 1]
```diff
@@ -35,9 +39,39 @@ export const ReportView: React.FC<Props> = ({
...
+  const navigate = useNavigate();
+
+  const save = async () => {
+    if (assignedAdmin.current === undefined) {
+      return;
+    }
+
+    try {
+      await assignAdmin(reportId, parseInt(assignedAdmin.current!.value));
+      navigate(0);
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:61]: this just reloads right? is there some benefit to this vs. using `window.location.reload()` (or whatever it is)? perhaps it keeps the react-router history? not sure but would love to know

File: src/components/admin/ReportView.tsx [hunk 2]
```diff
@@ -80,14 +114,32 @@ export const ReportView: React.FC<Props> = ({
           </ModalHeader>
           <ModalCloseButton></ModalCloseButton>
           <ModalBody fontSize="medium">
-            <HStack>
-              <Text fontWeight="semibold">Reason:</Text>
-              <Text>{reason}</Text>
-            </HStack>
-            <ReportObjectView type={type} object={associatedObject!} />
+            <VStack alignItems={'flex-start'} gap={3}>
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:117]: alignItems="flex-start"

File: src/services/report.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ const deserializeAssociatedObject = (
 function deserializedReport({
   report_id: reportId,
   reporter,
-  admin_id: adminId,
+  assignee,
```
Comment (elimelt) [inline src/services/report.ts:55]: this can be captured in `...rest` i think

File: src/services/report.ts [hunk 4]
```diff
@@ -133,3 +133,16 @@ export async function getReportDetail(reportId: string): Promise<Report> {
 
   return deserializedReport(res.data);
 }
+
+export async function assignAdmin(reportId: string, adminId: number) {
+  const url = `/reports/${reportId}/assign/`;
+  devPrint('called');
```
Comment (elimelt) [inline src/services/report.ts:139]: might want to remove

File: src/components/admin/ReportView.tsx (hunk 1 above)
Comment (Advayp) [reply src/components/admin/ReportView.tsx:61]: `navigate(0)` only reloads the current route, whereas `window.location.reload()` also reloads the assets. `navigate(0)` is more react-friendly and has less overhead.

File: src/services/report.ts (hunk 4 above)
Comment (elimelt) [reply src/services/report.ts:137]: btw best to reply if you choose not to

**Review by elimelt — APPROVED** [review]
lgtm

File: src/services/report.ts (hunk 4 above)
Comment (Advayp) [reply src/services/report.ts:137]: I thought I removed it? Let me double check

Comment (Advayp) [reply src/services/report.ts:137]: Oh I removed some other one I think. My bad.

**Pull request merged**